dask/tests/test_config.py

```diff
@@ -279,3 +279,18 @@ def test_refresh():
 
     refresh(paths=[], env={'DASK_C': '3'}, config=config, defaults=defaults)
     assert config == {'a': 1, 'c': 3}
+
+
+def test_expand_environment_variables():
+
+    os.environ["ENV_A"] = "foo"
+    config = '$ENV_A'
+    assert expand_environment_variables(config) == 'foo'
+
+    os.environ["ENV_B"] = "bar"
+    config = {'a': '$ENV_B'}
+    assert expand_environment_variables(config)['a'] == 'bar'
+
+    os.environ["ENV_C"] = "spam"
+    config = {'a': 'A', 'b': ['1', 2, '$ENV_C']}
+    assert expand_environment_variables(config)['b'] == ['1', 2, 'spam']
```

Comment:
We should probably clean up `os.environ` after running this test in a try-finally block.  Also, we can probably use `ENV_A` in all cases.